### PR TITLE
feat: Add comment-on-pr action from core

### PR DESCRIPTION
**Pull request by eKoopmans**
I'm working on making "PR Previews" into a more reusable pattern that all teams can easily implement. Part of that is moving the required actions out of core to somewhere more central.

This PR:

- [copies `comment-on-pr` from core](https://github.com/BrightspaceUI/core/blob/main/.github/actions/comment-on-pr/action.yml)
- updates inputs/outputs to lowercase
- adds a readme

I'll make a follow-up PR to core once this is merged, to use the shared action.

[US156216](https://rally1.rallydev.com/#/?detail=/userstory/719304972319&fdp=true)

### CODE REVIEW

File: comment-on-pr/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Comment on PR
+        uses: BrightspaceUI/actions/comment-on-pr@main
+        with:
+          MESSAGE: |
+            Thanks for the PR! 🎉
+
+            This is a sample multiline comment.
+          ONLY_POST_ONCE: true
```
Comment (svanherk) [inline comment-on-pr/README.md:28]:
```suggestion
          message: |
            Thanks for the PR! 🎉

            This is a sample multiline comment.
          only-post-once: true
```

File: comment-on-pr/action.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    description: A unique key attached invisibly to the comment, for use with the `only-post-once` option
+    default: gh-actions_comment-on-pr
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Find previous comment
+      if: ${{ inputs.only-post-once != 'false' }}
+      uses: Brightspace/third-party-actions@peter-evans/find-comment
```
Comment (eKoopmans) [inline comment-on-pr/action.yml:22]: @svanherk suggested moving off of this dependency, but I didn't see an easy way to do that. The closest I saw was [Octokit `issues.listComments`](https://octokit.github.io/rest.js/v19#issues-list-comments) - I could do that and then do a find in [the response](https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#list-issue-comments), searching the `body`.

Would probably work but I didn't want to fiddle with something that already works...

**Review by oelhanafey — APPROVED** [review]
LGTM, Gaudi will need to approve adding it

**Comment by oelhanafey** [comment]
Or not, I thought it would need a CODEOWNER approval. Might be good to wait for their approval anyway

**Review by dlockhart — APPROVED** [review]
Looks good -- thanks for production-izing this!

**Pull request merged**